src/SonarScanner.MSBuild.PreProcessor/PreProcessor.cs

```diff
@@ -27,181 +27,190 @@
+    }
 
-        public PreProcessor(IPreprocessorObjectFactory factory, ILogger logger)
+    public async Task<bool> Execute(IEnumerable<string> args)
+    {
+        logger.SuspendOutput();
+        var processedArgs = ArgumentProcessor.TryProcessArgs(args, logger);
+
+        if (processedArgs is null)
         {
-            this.factory = factory ?? throw new ArgumentNullException(nameof(factory));
-            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            logger.ResumeOutput();
+            logger.LogError(Resources.ERROR_InvalidCommandLineArgs);
+            return false;
         }
-
-        public async Task<bool> Execute(IEnumerable<string> args)
+        else
         {
-            logger.SuspendOutput();
-            var processedArgs = ArgumentProcessor.TryProcessArgs(args, logger);
-
-            if (processedArgs == null)
-            {
-                logger.ResumeOutput();
-                logger.LogError(Resources.ERROR_InvalidCommandLineArgs);
-                return false;
-            }
-            else
-            {
-                return await DoExecute(processedArgs);
-            }
+            return await DoExecute(processedArgs);
         }
+    }
 
-        private async Task<bool> DoExecute(ProcessedArgs localSettings)
+    private async Task<bool> DoExecute(ProcessedArgs localSettings)
+    {
+        Debug.Assert(localSettings is not null, "Not expecting the process arguments to be null");
+        logger.Verbosity = VerbosityCalculator.ComputeVerbosity(localSettings.AggregateProperties, logger);
+        logger.ResumeOutput();
+        InstallLoaderTargets(localSettings);
+        var buildSettings = BuildSettings.GetSettingsFromEnvironment();
+
+        // Create the directories
+        logger.LogDebug(Resources.MSG_CreatingFolders);
+        if (!Utilities.TryEnsureEmptyDirectories(logger, buildSettings.SonarConfigDirectory, buildSettings.SonarOutputDirectory))
         {
-            Debug.Assert(localSettings != null, "Not expecting the process arguments to be null");
-            logger.Verbosity = VerbosityCalculator.ComputeVerbosity(localSettings.AggregateProperties, logger);
-            logger.ResumeOutput();
-            InstallLoaderTargets(localSettings);
-            var buildSettings = BuildSettings.GetSettingsFromEnvironment();
+            return false;
+        }
 
-            // Create the directories
-            logger.LogDebug(Resources.MSG_CreatingFolders);
-            if (!Utilities.TryEnsureEmptyDirectories(logger, buildSettings.SonarConfigDirectory, buildSettings.SonarOutputDirectory))
+        using var server = await factory.CreateSonarWebServer(localSettings);
+        var jreResolver = factory.CreateJreResolver(server);
+        // TODO: This is expected to be the path to sonarUserHome, but it is not yet merged.
+        var resolvedJavaExePath = await jreResolver.ResolveJrePath(localSettings, "TODO");
```

Comment:
What do you think we should do with this @costin-zaharia-sonarsource ?
This depends on #2006, so for now I put a dummy value on the second parameter.

We either wait for sonar.userHome PR to be merged, or we can merge this and then do a follow-up PR.

Reply:
I don't see a problem with merging it as long we have a GitHub issue, to not forget, and finish the flow by the end of the sprint.